lib/JobProcessor.js

```diff
@@ -276,12 +277,25 @@ JobProcessor.prototype.jobProcessorAsync = function() {
             return;
         }
         // generate tile and repeat
-        return self.processOneTileAsync(iterValue).then(() => {
-            if (self.isShuttingDown) {
-                throw new Err('Shutting down');
-            }
-            return self.jobProcessorAsync();
-        });
+        // if timeout is 0 or not specified, e.g. null, keep old behavior
+        if (!this.tileTimeOut){
+            return self.processOneTileAsync(iterValue).then(() => {
+                if (self.isShuttingDown) {
+                    throw new Err('Shutting down');
+                }
+                return self.jobProcessorAsync();
+            });
+        } else {
+            // set timeout for tile process in case mapnik get stucked
+            return self.processOneTileAsync(iterValue).timeout(this.tileTimeOut).then(() => {
+                if (self.isShuttingDown) {
+                    throw new Err('Shutting down');
+                }
+                return self.jobProcessorAsync();
+            }).catch(Promise.TimeoutError, function(e) {
+                throw new Err("Tile processing timed out");
+            });;
```

Comment:
this statement realy really should end here :)

Reply:
Sorry, I didn't understand. Are you worried about the throw statement?

Reply:
you have two semicolons :)

Reply:
Done.